### PR TITLE
add expectedError method

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

File: src/log.js [hunk 1]
```diff
@@ -188,6 +189,20 @@ export class Log {
...
+  /**
+   * Reports an error message and marks with an expected property. If the
+   * logging is disabled, the error is rethrown asynchronously.
+   * @param {string} tag
+   * @param {...*} var_args
+   */
+  expectedError(tag, var_args) {
+    const error = this.error(tag, var_args);
+    if (error) {
+      error.expected = true;
```
Comment (dvoytenko) [inline src/log.js]: While correct, it's a bit weird to set flag on an error after it's been reported. Maybe better refactor call into a private method and call from both methods with different flags?

Comment (erwinmombay) [inline src/log.js]: yeah i can do that, that was actually the original code i had written.

**Comment by erwinmombay** [comment]
@dvoytenko changes made. PTAL

File: src/log.js [hunk 2]
```diff
@@ -179,14 +179,42 @@ export class Log {
    * asynchronously.
    * @param {string} tag
    * @param {...*} var_args
+   * @return {!Error|undefined}
    */
-  error(tag, var_args) {
+  error_(tag, var_args) {
```
Comment (jridgewell) [inline src/log.js]: Mark private.

File: src/log.js [hunk 3]
```diff
@@ -179,14 +179,42 @@ export class Log {
...
+   * Reports an error message and marks with an expected property. If the
+   * logging is disabled, the error is rethrown asynchronously.
+   * @param {string} tag
+   * @param {...*} var_args
+   */
+  expectedError(tag, var_args) {
+    const error = this.error_.apply(this, arguments);
+    if (error) {
+      error.expected = true;
       this.win.setTimeout(() => {throw error;});
```
Comment (jridgewell) [inline src/log.js]: Don't we have a `rethrowAsync`?

Comment (erwinmombay) [inline src/log.js]: yeah, i think it calls createErrorVarArgs again though

File: src/log.js (hunk 2 above)
Comment (erwinmombay) [inline src/log.js]: done.

File: src/log.js (hunk 3 above)
Comment (jridgewell) [inline src/log.js]: 👍

**Comment by erwinmombay** [comment]
@jridgewell can you point out to me what the expectedError is in xhr?

**Comment by jridgewell** [comment]
Ping.

**Comment by erwinmombay** [comment]
@jridgewell sorry bout that, will finish this today. got caught up in this other issue

File: src/service/xhr-impl.js [hunk 4]
```diff
@@ -394,7 +394,7 @@ function createXhrRequest(method, url) {
     xhr = new XDomainRequest();
     xhr.open(method, url);
   } else {
-    throw new Error('CORS is not supported');
+    throw dev().createExpectedError('CORS is not supported');
```
Comment (erwinmombay) [inline src/service/xhr-impl.js]: @dvoytenko can you review this?

File: src/service/xhr-impl.js [hunk 5]
```diff
@@ -365,10 +365,10 @@ export function fetchPolyfill(input, opt_init) {
       }
     };
     xhr.onerror = () => {
-      reject(new Error('Network failure'));
+      reject(user().createExpectedError('Network failure'));
     };
     xhr.onabort = () => {
-      reject(new Error('Request aborted'));
+      reject(user().createExpectedError('Request aborted'));
```
Comment (erwinmombay) [inline src/service/xhr-impl.js]: @dvoytenko please review

**Pull request merged**